src/api/code-challenge/content-types/code-challenge/lifecycles.js

```diff
@@ -1,43 +1,68 @@
+const { executeTests } = require("../code-challenge/test-executor");
 const { handleInternalLabel } = require("../../../../utils/general");
+const { compareIds } = require("../code-challenge/utils");
 
-// runMetaTestsNeed more specific name later, maybe use "validateInternalTests"
-async function runMetaTests(eventTests, eventMetaTests) {
+const runMetaTests = async ({ eventTests, eventMetaTests, challengeLabel }) => {
   const [internalTests, metaTests] = await Promise.all([
     getInternalTests(eventTests),
     getMetaTests(eventMetaTests),
   ]);
 
-  // console.log("final internalTests: ", internalTests);
-  // console.log("final metaTests: ", metaTests);
+  for (let i = 0; i < metaTests.length; i++) {
+    const metaCaseCode = metaTests[i].caseCode;
+    const metaLabel = metaTests[i].label;
+    const metaTestId = metaTests[i].id;
+    const evalResultShouldBe = metaTests[i].passes;
 
-  // For next part of feature: return some type of eval using internalTests & metaTests
-  return false;
-}
+    let metaTestResult = await executeTests({
+      metaCaseCode,
+      internalTests,
+      metaLabel,
+      challengeLabel,
+      metaTestId,
+      evalResultShouldBe,
+    });
+
+    // Feature: MetaTest Final Results being logged to Console.
+    console.log(
+      `\n\n<"${metaTests[i].label}" METATEST #${metaTests[i].id}> \n\n`,
+      metaTestResult,
+      "\n\n"
+    );
+  }
+};
 
 async function getInternalTests(eventTests) {
   const internalTests = await strapi.db
     .query("challenge.code-challenge-test")
     .findMany({
-      select: ["id", "internalTest"],
+      select: ["id", "internalTest", "label"],
       where: {
         id: eventTests.map(({ id: testId }) => testId),
       },
     });
-  return internalTests;
+
+  return internalTests.sort(compareIds);
 }
 
 async function getMetaTests(eventMetaTests) {
   const metaTests = await strapi.db.query("challenge.meta-test").findMany({
-    select: ["id", "caseCode", "passes"],
+    select: ["id", "caseCode", "passes", "label"],
     where: {
       id: eventMetaTests.map(({ id: metaId }) => metaId),
     },
   });
-  return metaTests;
+
+  return metaTests.sort(compareIds);
 }
 
 const beforeCreateOrUpdate = async (event) => {
-  runMetaTests(event.params.data.tests, event.params.data.MetaTest);
+  const eventParams = {
+    eventTests: event.params.data.tests,
+    eventMetaTests: event.params.data.MetaTest,
+    challengeLabel: event.params.data.internalLabel,
+  };
+  runMetaTests(eventParams);
```

Comment:
you don't need to have a separate line here, just do 
```
 runMetaTests({
    eventTests: event.params.data.tests,
    eventMetaTests: event.params.data.MetaTest,
    challengeLabel: event.params.data.internalLabel,
  });
```

Reply:
Thanks for the catch.